test/classes/delete-free/lifetimes/stecil-setup.chpl

```diff
@@ -8,22 +8,29 @@ class MyLocDom {
 
 class MyStencilDom {
   var A:[1..10] borrowed MyLocDom;
+  proc deinit() {
+    for i in 1..10 {
+      var c = _to_unmanaged(A[i]);
+      delete c;
+    }
+  }
 }
 
 config const branch = true;
 
 proc MyStencilDom.setup() {
   ref elem = A[1];
-  if branch then
+  if branch {
     elem = new unmanaged MyLocDom(1);
-  else
+  } else
     elem.x += 1;
 }
 
 proc test() {
   var dom = new unmanaged MyStencilDom();
   dom.setup();
   writeln(dom);
+  delete dom;
```

Comment:
If it's just for this test, I think it makes more sense to just delete dom.A[1] here just before the delete dom, but I'm also OK with what you wrote above (I just worry that it being in the deinit makes it look more general purpose than it is).

Reply:
So you're suggesting using `delete _to_unmanaged(dom.A[1]);` just above this line, is that right?

Reply:
Yes, exactly. BTW you should be able to write `delete dom.A[1]:unmanaged` (although parens might be required there somewhere).

Reply:
Oh, weird...  I'd originally tried casting to `unmanaged MyLocDom` to delete it, but that didn't work:

```
stecil-setup.chpl:27: error: illegal cast from MyLocDom to unmanaged MyLocDom
```

and it didn't occur to me to try just `: unmanaged`.  Is the error message from my attempt a bug or a feature?

Reply:
Probably a bug.

Reply:
See PR #12683.  I'll also open an issue if there isn't one already.